src/generator/writers/writeStdlib.ts

```diff
@@ -1426,4 +1426,24 @@ export function writeStdlib(ctx: WriterContext) {
             `);
         });
     });
+
+    ctx.fun(`__tact_pow`, () => {
+        ctx.signature(`int __tact_pow(int base, int exp)`);
+        ctx.context("stdlib");
+        ctx.body(() => {
+            ctx.write(`
+                int result = 1;
+                repeat (exp) {
+                    result *= base;
+                }
+                return result;
```

Comment:
Would [exponentiation by squaring](https://en.wikipedia.org/wiki/Exponentiation_by_squaring) be more gas efficient?

Reply:
yes, right. but only for `exp` larger than some X.
maybe I should do some benchmarks on that and add a simple `if-else` depending on `exp` with two different implementations?

Reply:
right, benchmarking sounds like a great idea, if for smaller exponents the gas difference is not that big we can use the fast exp algorithm without a condition

Reply:
I've done some benchmarking and have come to the conclusion that in case with smart contracts and integer numbers, it's more efficient to leave the `pow` calculation as it is currently done in this PR. The reason is that, due to integer number limits and the absence of real values, we will never need to calculate `pow` with an `exp` large enough to outperform a simple loop using such optimizations.